### PR TITLE
Update ruff linter / formatter

### PR DESCRIPTION
**Pull request by jarrodmillman**
## Description

I noticed (using repo-review) a syntax error in pyproject.toml and then noticed we hadn't enabled ruff formatting.  I reviewed the changes and they are all reasonable. I would like to get this merged before he 0.25 release, since we had a syntax error in our pyproject.toml. This should be merged (and not squashed) to preserve the git blame ignore commit hash.

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpydoc/blob/main/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

For maintainers and optionally contributors, please refer to the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) on how to document this PR for the release notes.

```release-note
...
```

### CODE REVIEW

**Review by lagru — APPROVED** [review]

File: doc/examples/filters/plot_nonlocal_means.py [hunk 1]
```diff
@@ -51,7 +51,9 @@
 print(f'estimated noise standard deviation = {sigma_est}')
 
 patch_kw = dict(
-    patch_size=5, patch_distance=6, channel_axis=-1  # 5x5 patches  # 13x13 search area
+    patch_size=5,
+    patch_distance=6,
+    channel_axis=-1,  # 5x5 patches  # 13x13 search area
```
Comment (stefanv) [inline doc/examples/filters/plot_nonlocal_means.py:56]: These should go next to `patch_size` and `patch_distance`

File: skimage/_vendored/numpy_lookfor.py [hunk 2]
```diff
@@ -132,7 +132,9 @@ def _lookfor_generate_cache(module, import_modules, regenerate):
             for n, v in _getmembers(item):
                 try:
                     item_name = getattr(
-                        v, '__name__', "%s.%s" % (name, n)  # noqa: UP031
+                        v,
+                        '__name__',
+                        "%s.%s" % (name, n),  # noqa: UP031
```
Comment (stefanv) [inline skimage/_vendored/numpy_lookfor.py:137]: Wonder why we have `noqa` here?

Comment (jarrodmillman) [reply skimage/_vendored/numpy_lookfor.py:137]: I was planning to do a search for `noqa` once he release is out. I will make a follow-up PR incorporating your comments.

File: doc/gh-pages.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 that is how the resulting directory will be named.
 
 In practice, you should use either actual clean tags from a current build or
-something like 'current' as a stable URL for the mest current version of the """
+something like 'current' as a stable URL for the mest current version of the"""
```
[the remark this answers is not in the document]
Comment (jarrodmillman) [reply doc/gh-pages.py:12]: It looks like we can just remove the file.

**Pull request merged**